library/swift/src/filters/AsyncRequestFilter.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+/// RequestFilter supporting asynchronous resumption.
+public protocol AsyncRequestFilter: RequestFilter {
+  /// Invoked explicitly in response to an asynchronous resumeRequest() callback when filter
+  /// iteration has been stopped. The parameters passed to this invocation will be a snapshot
+  /// of any stream state that has not yet been forwarded along the filter chain.
+  ///
+  /// - return: The resumption status including any HTTP entities that will be forwarded.
+  func onResumeRequest(
+    headers: RequestHeaders?,
+    data: Data?,
+    trailers: RequestTrailers?,
+    endStream: Bool
```

Comment:
When will `endStream` be true? Will it be true if *either* there's a data frame with `endStream=true` *or* trailers?

Reply:
I think it's worth adding `- parameter` docs for each of these (same with the response filter) to clarify these points